Msi 2287 in store pickup orders placement from admin panel

novikor opened the pull request:
<!---
    Thank you for contributing to Magento.
    To help us process this pull request we recommend that you add the following information:
     - Summary of the pull request,
     - Issue(s) related to the changes made,
     - Manual testing scenarios
    Fields marked with (*) are required. Please don't remove the template.
-->

<!--- Please provide a general summary of the Pull Request in the Title above -->

### Description (*)
Well, see the initial issue.
At the moment, a dropdown has been created for the order creation page.
![image](https://user-images.githubusercontent.com/26086561/61002060-787f1a00-a369-11e9-9bf6-3c1e5b4e1909.png)


### Fixed Issues (if relevant)
<!---
    If relevant, please provide a list of fixed issues in the format magento/magento2#<issue_number>.
    There could be 1 or more issues linked here and it will help us find some more information about the reasoning behind this change.
-->
1. magento/inventory#2287:In-Store Pickup orders placement from Admin Panel

### Manual testing scenarios (*)
Not today 

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

Commit:
MSI-2286-In-Store-Pickup-checkout-implementation-on-Weabpi-level.
Replace Pickup Location address for Shipping address during checkout flow.

novikor committed:
MSI-2332 In-Store Pickup orders placement from Admin Panel #2287.
Initial point of the feature.
Displayed the dropdown with sources list on order create page.

ishakhsuvarov commented on app/code/Magento/InventoryInStorePickupAdminUi/Model/GetPickupSources.php:
```diff
@@ -0,0 +1,74 @@
+            )->getStockId()
+        );
+
+        return array_filter($stockSources, [$this->getIsPickupLocationActive, 'execute']);
```
That looks like we need to rethink some of the services

ishakhsuvarov commented on app/code/Magento/InventoryInStorePickupAdminUi/ViewModel/CreateOrder/SourcesForm.php:
```diff
@@ -0,0 +1,92 @@
+    public function getPickupLocationsArray(): array
+    {
+        if ($this->pickupLocationsArray === null) {
+            $this->initializePickupLocationsArray();
```
Even though it seems to be a part of in-object-cache currently naming looks like it does some kind of init. I'd suggest changing it to something like load

novikor replied:
`initializePickupLocationsArray` -> `loadPickupLocationsArray` ?

ishakhsuvarov replied:
Do we need "Array" in there?

novikor replied:
I don`t like this, too. Probably "Options" would look better

novikor replied:
`getPickupLocationsArray` and `loadPickupLocations`

ishakhsuvarov commented on app/code/Magento/InventoryInStorePickupAdminUi/view/adminhtml/templates/order/create/shipping/method/sources_form.phtml:
```diff
@@ -0,0 +1,38 @@
+
+// @codingStandardsIgnoreFile
+/** @var SourcesForm $viewModel */
+$viewModel = $block->getData('view_model');
```
Let's go with the conventional `$block->getViewModel()`

novikor replied:
Just don`t like magic. Fixed.

ishakhsuvarov commented on app/code/Magento/InventoryInStorePickupAdminUi/view/adminhtml/templates/order/create/shipping/method/sources_form.phtml:
```diff
@@ -0,0 +1,38 @@
+>
+    <div class="admin__field _required">
+        <label class="label admin__field-label" for="pickup_location_source">
+            <span><?= $block->escapeHtml(__('Pickup Location Source')) ?></span>
```
Would be nice not to forget discussing naming for the Admin UI

swnsma replied:
@novikor `Pickup Location Source` or `Pickup Source` or ... ?

novikor replied:
Yes

ishakhsuvarov commented on app/code/Magento/InventoryInStorePickupAdminUi/Model/GetPickupSources.php:
```diff
@@ -0,0 +1,74 @@
+     * @throws LocalizedException
+     * @throws NoSuchEntityException
+     */
+    public function execute(string $websiteCode): array
```
Should we stick to the convention of sales channel and code? Just to make sure that those customizing default sales channel would have less work?

novikor replied:
Do you mean it would be better to use SalesChannel/Stock as arguments instead of `$websiteCode`?

ishakhsuvarov replied:
I'm pretty sure it would. Seems to create less customization points for someone doing non-default sales channels

novikor replied:
Done

ishakhsuvarov commented on app/code/Magento/InventoryInStorePickupAdminUi/Model/GetPickupSources.php:
```diff
@@ -0,0 +1,74 @@
+     */
+    public function execute(string $websiteCode): array
+    {
+        $stockSources = $this->getSourcesAssignedToStockOrderedByPriority->execute(
```
Might not need priority ordering here. However it's a nice touch for now.

novikor replied:
It`s just the only service that provides sources by stock I found :)

ishakhsuvarov reviewed (commented):
Looks cool so far

novikor commented:
Have no idea why GitHub shows @swnsma commits despite they are already present in the base branch.

Commit:
MSI-2287 in store pickup orders placement from admin panel #2390.
Pickup location saving via the admin panel

Commit:
MSI-2287 in store pickup orders placement from admin panel #2390.
Set custom shipping description including pickup location name
for In-Store pickup orders

Commit:
MSI-2287 in store pickup orders placement from admin panel #2390.
Some refactoring

smoskaluk commented:
Hi @novikor, thank you for your contribution, but it looks like parts
- Shipping Address edit form should be disabled (or ignored) if selected Delivery Method 'In-Store Pickup'.
- Possibility for Shipping Address 'the same as Billing' should be disabled for In-Store Pickup Orders. 
not implemented in this PR, or I missed something? 
Please, advise.

novikor commented:
Hi, @smoskaluk .
Thank you for testing.
What about point 1: the PR is still in progress, so has a "draft" status. I will implement this feature in the nearest future.
What about point 2: I will check it, too. Probably will create a separate issue for that.

Commit:
Merge remote-tracking branch 'origin/MSI-2286-In-Store-Pickup-checkout-implementation-on-Weabpi-level' into MSI-2287-In-Store-Pickup-orders-placement-from-Admin-Panel

nmalevanec committed:
Store pickup order placement from admin panel with Simple product automation.

novikor committed:
MSI-2287 in store pickup orders placement from admin panel #2390.
Fixed replacing order shipping address for adminhtml.
Added TODO-notice.

ishakhsuvarov commented:
@magento run all tests

Commit:
MSI-2287 in store pickup orders placement from admin panel #2390.
Initial phase of disabling shipping address form

Commit:
Merge remote-tracking branch 'origin/store-pickup' into MSI-2287-In-Store-Pickup-orders-placement-from-Admin-Panel

Commit:
MSI-2287 in store pickup orders placement from admin panel #2390.
Removed abandoned plugin.

Commit:
MSI-2287 in store pickup orders placement from admin panel #2390.
\Magento\InventoryInStorePickupQuote\Model\ToQuoteAddress redesign: Use address data array instead of pickup location itself.

Commit:
MSI-2287 in store pickup orders placement from admin panel #2390.
Finally made possible to place order using Store Pickup.
Replace shipping address with data from Source.

TBlindaruk commented on app/code/Magento/InventoryInStorePickupAdminUi/view/adminhtml/web/order/create/trigger-shipping-method-update.js:
```diff
@@ -0,0 +1,19 @@
+    function ($) {
+        'use strict';
+        return function () {
+        debugger;
```
=_=

ishakhsuvarov committed:
Static test failure fixes

ishakhsuvarov committed:
Merge branch 'store-pickup' of github.com:magento/inventory into MSI-2287-In-Store-Pickup-orders-placement-from-Admin-Panel

ishakhsuvarov committed:
Coding style adjustments

Commit:
Merge remote-tracking branch 'origin/store-pickup' into MSI-2287-In-Store-Pickup-orders-placement-from-Admin-Panel

magento-cicd2 commented:
[![CLA assistant check](https://cla.dev.magento.com/pull/badge/signed)](https://cla.dev.magento.com/magento/inventory?pullRequest=2390) <br/>All committers have signed the CLA.

Commit:
MSI-2287 in store pickup orders placement from admin panel #2390.

Commit:
MSI-2287 in store pickup orders placement from admin panel #2390.
Hide sources form if store pickup is not selected

novikor commented:
Ready for review.
If tests do not fail.

novikor commented:
I`ll try to fix the functional tests one day.

swnsma commented:
Hi @novikor!
Could you solve issue with the CLA?

swnsma commented on app/code/Magento/InventoryInStorePickupAdminUi/Model/ExtractSourceShippingAddressData.php:
```diff
@@ -0,0 +1,62 @@
+     * Extract Shipping Address fields from Source.
+     *
+     * @TODO Refactor when issue will be resolved in core.
+     * @see Please check issue in core for more details: https://github.com/magento/magento2/issues/23386.
```
Already may be refactored.

novikor replied:
Not now

swnsma commented on app/code/Magento/InventoryInStorePickup/Model/PickupLocation/GetPickupLocationByCode.php:
```diff
@@ -0,0 +1,56 @@
+<?php
```
I think we does not need one more service for these purposes.
We already have a service which can provide Pickup Location based on the code and Sales Channel.

novikor replied:
ok

swnsma commented on app/code/Magento/InventoryInStorePickupAdminUi/Plugin/Sales/Model/AdminOrder/Create/SetPickupLocationFromPost.php:
```diff
@@ -0,0 +1,86 @@
+    }
+
+    /**
+     * Add pickup location code to the shipping address
```
But this is Source Code...

novikor replied:
No. I use `\Magento\InventoryInStorePickupQuote\Model\Address\SetAddressPickupLocation` there

swnsma commented on app/code/Magento/InventoryInStorePickupAdminUi/Plugin/Sales/Model/AdminOrder/Create/SetPickupLocationFromPost.php:
```diff
@@ -0,0 +1,86 @@
+
+declare(strict_types=1);
+
+namespace Magento\InventoryInStorePickupAdminUi\Plugin\Sales\Model\AdminOrder\Create;
```
Please change namespace with mention of corresponding class which it intercepts.
This class has all indications to be a part of Presentation Layer.

novikor replied:
Ok

swnsma commented on app/code/Magento/InventoryInStorePickupAdminUi/ViewModel/CreateOrder/SourcesForm.php:
```diff
@@ -0,0 +1,110 @@
+    /**
+     * Get list of inventory sources assigned as pickup locations.
+     *
+     * @return array [sourceCode => SourceName]
```
So, `Studly Caps` on `Camel Case`?

novikor replied:
camelCase

swnsma commented on app/code/Magento/InventoryInStorePickupAdminUi/view/adminhtml/templates/order/create/storepickup-script-load.phtml:
```diff
@@ -0,0 +1,20 @@
+            }
+        }
+    }
+</script>
```
Missed empty line in the end of the file.

novikor replied:
Fixed

swnsma commented on app/code/Magento/InventoryInStorePickupQuote/Model/Address/SetAddressPickupLocation.php:
```diff
@@ -0,0 +1,49 @@
+
+/**
+ * Set Address pickup location
+ * @api
```
We don't want to expose any API for `InventoryInStorePickupQuote`.

novikor replied:
OK

swnsma commented on app/code/Magento/InventoryInStorePickupQuote/Model/ToQuoteAddress.php:
```diff
@@ -9,13 +9,14 @@
 /**
  * Convert Pickup Location and provided Shipping Address to Pickup Location Quote Address.
+ *
+ * @api
```
We don't want to expose any API for InventoryInStorePickupQuote.

novikor replied:
ok

swnsma commented on app/code/Magento/InventoryInStorePickupQuote/composer.json:
```diff
@@ -4,10 +4,12 @@
     "magento/module-quote": "*",
     "magento/module-inventory-in-store-pickup-shipping-api": "*",
     "magento/module-store": "*",
+    "magento/module-checkout": "*",
```
This is REALLY something we would like to avoid.
Could you re-arrange extensions of order flow to remove dependency on `Magento_Checkou` ?

novikor replied:
Agree. Fixed

swnsma commented on app/code/Magento/InventoryInStorePickupShipping/composer.json:
```diff
@@ -7,6 +7,7 @@
         "magento/module-inventory-in-store-pickup-shipping-api": "*",
         "magento/module-inventory-in-store-pickup-api": "*",
         "magento/module-inventory-sales-api": "*",
+        "magento/module-inventory-in-store-pickup-quote": "*",
```
Could we somehow remove this dependency?

I prefer to avoid such dependencies:
`magento/module-inventory-in-store-pickup-quote` -> `magento/module-inventory-in-store-pickup-shipping-api`
`magento/module-inventory-in-store-pickup-shipping` -> `magento/module-inventory-in-store-pickup-shipping-api`
`magento/module-inventory-in-store-pickup-shipping` -> `magento/module-inventory-in-store-pickup-quote`

novikor replied:
ok

swnsma commented on app/code/Magento/InventoryInStorePickupQuote/Plugin/Checkout/AddPickupLocationAddressToShippingInformation.php:
```diff
@@ -0,0 +1,144 @@
+        int $cartId,
+        ShippingInformationInterface $addressInformation
+    ) {
+        $shippingCarrierCode = $addressInformation->getShippingCarrierCode();
```
Very difficult to read the method.
Could you please refactor it or split to smaller parts to improve readability?

novikor replied:
...

swnsma reviewed (changes requested):
Please check my comments.

Commit:
MSI-2287 in store pickup orders placement from admin panel #2390.
Some bug and typos fixes

ishakhsuvarov committed:
Merge branch 'MSI-2399' of github.com:magento/inventory into MSI-2287-In-Store-Pickup-orders-placement-from-Admin-Panel

ishakhsuvarov committed:
Merge branch 'store-pickup' of github.com:magento/inventory into MSI-2287-In-Store-Pickup-orders-placement-from-Admin-Panel

Conflicts:
	app/code/Magento/InventoryInStorePickupAdminUi/composer.json
	app/code/Magento/InventoryInStorePickupAdminUi/i18n/en_US.csv
	app/code/Magento/InventoryInStorePickupQuote/Model/ToQuoteAddress.php
	app/code/Magento/InventoryInStorePickupQuote/Plugin/Quote/ReplaceShippingAddressForShippingAddressManagement.php

Commit:
MSI-2287 in store pickup orders placement from admin panel #2390.
Code review improvements

novikor commented:
@swnsma, please review

Commit:
MSI-2287 in store pickup orders placement from admin panel #2390.
Code review improvements

swnsma commented on app/code/Magento/InventoryInStorePickupAdminUi/Model/GetPickupSources.php:
```diff
@@ -0,0 +1,68 @@
+    public function __construct(
+        GetSourcesAssignedToStockOrderedByPriorityInterface $getSourcesAssignedToStockOrderedByPriority
+    ) {
+
```
Remove empty line please.

novikor replied:
ok

swnsma commented on app/code/Magento/InventoryInStorePickup/Model/ExtractSourceShippingAddressData.php:
```diff
@@ -0,0 +1,62 @@
+/**
+ * Extract Shipping Address fields from Source.
+ */
+class ExtractSourceShippingAddressData
```
Service should extract address data. Not Shipping Address Data.
Shipping Address should be extracted in Quote module.

swnsma commented on app/code/Magento/InventoryInStorePickupShippingApi/Model/Carrier/GetCarrierTitle.php:
```diff
@@ -0,0 +1,49 @@
+ *
+ * @api
+ */
+class GetCarrierTitle
```
Can we use this service in \Magento\InventoryInStorePickupShippingApi\Model\Carrier\InStorePickup::createResultMethod
To make consistent receiving of carrier title?

novikor replied:
ok

swnsma commented on app/code/Magento/InventoryInStorePickupQuote/Plugin/Quote/Address/SetShippingDescription.php:
```diff
@@ -0,0 +1,117 @@
+namespace Magento\InventoryInStorePickupQuote\Plugin\Quote\Address;
+
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\InventoryInStorePickup\Model\GetPickupLocation;
```
Please use abstraction instead of implementation.

novikor replied:
ok

swnsma commented on app/code/Magento/InventoryInStorePickupQuote/Plugin/Quote/Address/SetShippingDescription.php:
```diff
@@ -0,0 +1,117 @@
+
+use Magento\Framework\Exception\NoSuchEntityException;
+use Magento\InventoryInStorePickup\Model\GetPickupLocation;
+use Magento\InventoryInStorePickup\Model\PickupLocation\GetPickupLocationByCode;
```
Do we still have this service?

novikor replied:
nope

swnsma commented on app/code/Magento/InventoryInStorePickupAdminUi/Controller/Adminhtml/Order/NotifyPickup.php:
```diff
@@ -23,7 +23,7 @@
  * Notify Customer of order pickup availability.
  */
-class NotifyPickup extends Action implements HttpPostActionInterface
+class NotifyPickup extends Action implements HttpGetActionInterface
```
It seems to be POST action, not GET.
Could you fix it or create new issue to investigate?

novikor replied:
No - it is not POST

swnsma replied:
You do not receive data.
You just make an action.
Why it is not 'POST' ?

ishakhsuvarov replied:
Seems more like POST to me

novikor replied:
There is no choice. See the ship button.

swnsma commented on app/code/Magento/InventoryInStorePickup/composer.json:
```diff
@@ -6,6 +6,7 @@
         "magento/framework": "*",
         "magento/module-inventory-in-store-pickup-api": "*",
         "magento/module-inventory-distance-based-source-selection-api": "*",
+        "magento/module-inventory-in-store-pickup-shipping-api": "*",
```
It is not looking as proper dependency.

novikor replied:
It will be refactored not in the scope of this issue

swnsma commented on app/code/Magento/InventoryAdminUi/Test/Mftf/Data/SourceExtensionAttributeData.xml:
```diff
@@ -0,0 +1,17 @@
+<entities xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+          xsi:noNamespaceSchemaLocation="urn:magento:mftf:DataGenerator/etc/dataProfileSchema.xsd">
+    <entity name="StorePickupEnabled" type="source_extension_attribute">
+        <data key="is_pickup_location_active">true</data>
```
Maybe these changes need to be placed in `app/code/Magento/InventoryInStorePickupAdminUi/Test/Mftf/Data/SourceExtensionAttributeData.xml` ?

novikor replied:
Yes

swnsma reviewed (changes requested):
Please check my notices.

Commit:
MSI-2287 in store pickup orders placement from admin panel #2390.
Moved MFTF to the correct module

Commit:
MSI-2287 in store pickup orders placement from admin panel #2390.
CR changes

Commit:
MSI-2287 in store pickup orders placement from admin panel #2390.
CR changes

ishakhsuvarov commented on app/code/Magento/InventoryAdminUi/Test/Mftf/Metadata/source_extension_attribute-meta.xml:
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Let's move this to `InventoryStorePickupAdminUi`.

ishakhsuvarov commented on app/code/Magento/InventoryAdminUi/Test/Mftf/Section/AdminOrderDetailsMainActionsSection.xml:
```diff
@@ -0,0 +1,14 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Let's move this to `InventoryStorePickupAdminUi`.

ishakhsuvarov commented on app/code/Magento/InventoryAdminUi/Test/Mftf/Section/AdminOrderFormPaymentSection.xml:
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Let's move this to `InventoryStorePickupAdminUi`.

ishakhsuvarov commented on app/code/Magento/InventoryAdminUi/Test/Mftf/Section/AdminOrderFormPaymentSection.xml:
```diff
@@ -0,0 +1,15 @@
+<sections xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="urn:magento:mftf:Page/etc/SectionObject.xsd">
+    <section name="AdminOrderFormPaymentSection">
+        <element name="inStorePickupOption" type="radio" selector="#s_method_in_store_pickup" timeout="30"/>
```
There might be a typo either in test or on the page: `#s_method_in_store_pickup`

ishakhsuvarov replied:
Irrelevant

ishakhsuvarov commented on app/code/Magento/InventoryInStorePickup/Model/ExtractPickupLocationShippingAddressData.php:
```diff
@@ -39,20 +37,10 @@ public function __construct(Copy $copyService)
     public function execute(PickupLocationInterface $pickupLocation): array
     {
-        $pickupLocationAddressData = $this->objectCopyService->getDataFromFieldset(
+        return $this->objectCopyService->getDataFromFieldset(
```
We might as well use object copy service directly where needed.

ishakhsuvarov commented on app/code/Magento/InventoryInStorePickup/Model/ExtractSourceShippingAddressData.php:
```diff
@@ -0,0 +1,50 @@
+     */
+    public function execute(SourceInterface $source): array
+    {
+        return $this->objectCopyService->copyFieldsetToTarget(
```
We might as well consider using it directly

ishakhsuvarov commented on app/code/Magento/InventoryInStorePickupAdminUi/Model/GetPickupSources.php:
```diff
@@ -0,0 +1,60 @@
+    }
+
+    /**
+     * Get list of sources marked as pickup location by website.
```
Looks like not by website but by stock

swnsma commented on app/code/Magento/InventoryInStorePickup/Model/ExtractPickupLocationShippingAddressData.php:
```diff
@@ -5,12 +5,10 @@
  */
 declare(strict_types=1);
 
-namespace Magento\InventoryInStorePickupQuote\Model;
```
Maybe it is better to call 'ExtractPickupLocationAddressData' ?

swnsma commented on app/code/Magento/InventoryInStorePickup/Model/ExtractSourceShippingAddressData.php:
```diff
@@ -0,0 +1,50 @@
+    /**
+     * Extract Shipping Address fields from Source.
+     *
+     * @TODO Refactor when issue will be resolved in core.
```
Please refactor.
see app/code/Magento/InventoryInStorePickup/Model/ExtractPickupLocationShippingAddressData.php

swnsma commented on app/code/Magento/InventoryInStorePickupQuote/Model/BuildShippingAddressData.php:
```diff
@@ -0,0 +1,37 @@
+/**
+ * Finalize shipping address data for the Store Pickup cart.
+ */
+class BuildShippingAddressData
```
I do not like word 'Build' in this context.

novikor replied:
Resolved

Commit:
MSI-2287 in store pickup orders placement from admin panel #2390.
CR changes

Commit:
Merge remote-tracking branch 'origin/store-pickup' into MSI-2287-In-Store-Pickup-orders-placement-from-Admin-Panel

Commit:
MSI-2287 in store pickup orders placement from admin panel #2390.
CR changes

Commit:
1720: Fix allows you to specify invalid parameters

Commit:
MSI-2287 in store pickup orders placement from admin panel #2390.
Post-merge bug fixes

Commit:
MSI-2287 in store pickup orders placement from admin panel #2390.
Fixed frontend name saving

Commit:
MSI-2287 in store pickup orders placement from admin panel #2390.
Fixed static tests

swnsma commented on app/code/Magento/InventoryInStorePickupAdminUi/Model/GetPickupSources.php:
```diff
@@ -0,0 +1,60 @@
+        return array_filter(
+            $stockSources,
+            function (SourceInterface $source): bool {
+                $extension = $source->getExtensionAttributes();
```
```suggestion
                return $source->getExtensionAttributes() && $source->getExtensionAttributes()->getIsPickupLocationActive();
```

swnsma commented on app/code/Magento/InventoryInStorePickupAdminUi/Model/GetShippingAddressBySourceCodeAndOriginalAddress.php:
```diff
@@ -0,0 +1,79 @@
+        ToQuoteAddress $sourceToQuoteAddress,
+        GetShippingAddressData $getShippingAddressData
+    ) {
+        $this->sourceRepository = $sourceRepository;
```
It seems that not all dependencies are used in the class.

ishakhsuvarov replied:
Fixed

swnsma commented on app/code/Magento/InventoryInStorePickupQuote/etc/fieldset.xml:
```diff
@@ -10,127 +10,127 @@
         <fieldset id="inventory_convert_pickup_location">
             <field name="country_id">
-                <aspect name="to_pickup_location_shipping_address"/>
+                <aspect name="to_store_pickup_shipping_address"/>
```
I am not sure, but probably `to_in_store_pickup_shipping_address` ?

novikor replied:
probably, but I don`t think it matters

swnsma replied:
Devil in details. It is better to call functionality as 'In-Store Pickup' everywhere.
Please change.

ishakhsuvarov replied:
Fixed

swnsma commented on app/code/Magento/InventoryInStorePickupShippingApi/Model/Carrier/GetCarrierTitle.php:
```diff
@@ -0,0 +1,49 @@
+     */
+    public function execute(?int $storeId = null): string
+    {
+        return $storeId ?
```
Isn't that normal behavior?

novikor replied:
If no `storeId` is passed - yes.

novikor replied:
If it is passed - yes

novikor replied:
What`s the question?

swnsma replied:
What the difference if you just leave it as:
```
$this->scopeConfig->getValue(self::CONFIG_PATH, ScopeInterface::SCOPE_STORE, $storeId);
```
?

ishakhsuvarov commented:
@magento run Functional Tests build

novikor commented:
@magento run Functional Tests build

swnsma commented on app/code/Magento/InventoryInStorePickupQuote/Plugin/Quote/Address/SetShippingDescription.php:
```diff
@@ -0,0 +1,114 @@
+        Quote $quote
+    ) {
+        $address = $quote->getShippingAddress();
+        if ($address->getShippingMethod() == InStorePickup::DELIVERY_METHOD
```
```suggestion
        if ($address->getShippingMethod() === InStorePickup::DELIVERY_METHOD
```

novikor replied:
For what purpose?

swnsma replied:
Just in case

novikor replied:
Nope

ishakhsuvarov replied:
Yes. We must take every action to make code robust and future-proof

swnsma commented on app/code/Magento/InventoryInStorePickupAdminUi/Model/GetShippingAddressBySourceCodeAndOriginalAddress.php:
```diff
@@ -0,0 +1,79 @@
+     * @return AddressInterface|null
+     * @throws NoSuchEntityException
+     */
+    public function execute(string $sourceCode, AddressInterface $originalAddress): ?AddressInterface
```
I am not sure, but, probably, it could be unified interface for the Quote to receive address from PickupLocation/Source.
We should think about this during dependencies segregation.

ishakhsuvarov replied:
https://github.com/magento/inventory/issues/2625

swnsma commented on app/code/Magento/InventoryInStorePickupAdminUi/Model/GetShippingAddressBySourceCodeAndOriginalAddress.php:
```diff
@@ -0,0 +1,79 @@
+    }
+
+    /**
+     * Get Shipping Address from Source by its sourceCode and original Shipping Address
```
```suggestion
     * Get Shipping Address from Source by its source code and original Shipping Address
```

slavvka commented:
@magento run Functional Tests build

ishakhsuvarov committed:
magento/inventory#2576: Refactored to reduce object coupling and follow backward compatibility policy

ishakhsuvarov committed:
Merge branch '2.3-develop' of github.com:magento/inventory into 1720-Fixes-Api

ishakhsuvarov committed:
magento/inventory#2576: Fixed fallback for products with some sources set for notify qty

ishakhsuvarov committed:
magento/inventory#2576: Update Code Style

novikor committed:
Update app/code/Magento/InventoryInStorePickupAdminUi/Model/GetShippingAddressBySourceCodeAndOriginalAddress.php

Co-Authored-By: Oleksandr Kravchuk <swnsma@gmail.com>

Nazar65 committed:
Increase MFTF tests coverage

ishakhsuvarov committed:
magento/inventory#2576: Update test data providers to take into account changes in fixtures

novikor committed:
Update app/code/Magento/InventoryInStorePickupAdminUi/Model/GetPickupSources.php

Co-Authored-By: Oleksandr Kravchuk <swnsma@gmail.com>

Commit:
Merge remote-tracking branch 'origin/store-pickup' into MSI-2287-In-Store-Pickup-orders-placement-from-Admin-Panel

nmalevanec committed:
Unskip Web-Api PreventDefaultSourceDisablingTest.

Commit:
MSI-2287 in store pickup orders placement from admin panel #2390.
Fixed an ultra-bug

ishakhsuvarov committed:
Merge branch '2.3-develop' into 1720-Fixes-Api

Commit:
MSI-2287 in store pickup orders placement from admin panel #2390.
MFTF: InStorePickupOrderPlacementWithSimpleProductAdditionalStockTest

Commit:
Merge remote-tracking branch 'origin/store-pickup' into MSI-2287-In-Store-Pickup-orders-placement-from-Admin-Panel

Stepa4man committed:
MSI-2599: InventoryConfigurableProductIndexer uses all columns value in predicate for clearing index

ishakhsuvarov committed:
Merge pull request #2576 from larsroettig/1720-Fixes-Api

#1720: Fix allows you to specify invalid parameters

Stepa4man committed:
MSI-2599: Fix static tests

Stepa4man committed:
Merge branch '2.3-develop' into MSI-2599-clearing-index-with-wrong-predicate

Nazar65 committed:
Increase MFTF tests coverage

Nazar65 committed:
Merge branch '2.3-develop' of https://github.com/magento/inventory into msi-mftf-coverage-6

nmalevanec committed:
Increase MFTF tests coverage.

ishakhsuvarov committed:
Merge pull request #2608 from Nazar65/msi-mftf-coverage-6

Msi mftf coverage 6

ishakhsuvarov committed:
Merge pull request #2610 from nmalevanec/MFTF-Tests

Increase MFTF tests coverage.

nmalevanec committed:
Increase MFTF tests coverage.

ishakhsuvarov committed:
Merge pull request #2612 from nmalevanec/MFTF-Tests

Increase MFTF tests coverage.

nmalevanec committed:
Increase MFTF tests coverage.

nmalevanec committed:
Increase MFTF tests coverage.

Stepa4man committed:
MSI-2599: Fix static tests

Stepa4man committed:
MSI-2599: Added test

ishakhsuvarov committed:
Merge branch '2.3-develop' of github.com:magento/inventory into MSI-2599-clearing-index-with-wrong-predicate

ishakhsuvarov committed:
magento/inventory#2603: Coding style fixes

ishakhsuvarov committed:
magento/inventory#2613: Update test modes

ishakhsuvarov committed:
Merge branch '2.3-develop' into MSI-2326

filmaj committed:
📝 update readme with better link for slack signup

ishakhsuvarov committed:
Merge pull request #2603 from magento/MSI-2599-clearing-index-with-wrong-predicate

MSI-2599: Clearing index with wrong predicate

ishakhsuvarov committed:
Merge pull request #2615 from magento/filmaj-slack-readme

📝 update readme with better link for slack signup

nmalevanec committed:
MSI-2616: Can't create shipment for order with grouped product on custom stock.
Fixed getting incorrect product type for grouped product.

duxabilii committed:
Update db_schema.xml

When importing, some city names are truncated due to the short string length. 
According to Readme.txt inside of exported archive

The data format is tab-delimited text in utf8 encoding, with the following fields :

country code      : iso country code, 2 characters
postal code       : varchar(20)
place name        : varchar(180)
admin name1       : 1. order subdivision (state) varchar(100)
admin code1       : 1. order subdivision (state) varchar(20)
admin name2       : 2. order subdivision (county/province) varchar(100)
admin code2       : 2. order subdivision (county/province) varchar(20)
admin name3       : 3. order subdivision (community) varchar(100)
admin code3       : 3. order subdivision (community) varchar(20)
latitude          : estimated latitude (wgs84)
longitude         : estimated longitude (wgs84)
accuracy          : accuracy of lat/lng from 1=estimated, 4=geonameid, 6=centroid of addresses or shape

sidolov committed:
Merge pull request #2613 from nmalevanec/B2b-MFTF-Tests

Increase MFTF tests coverage.

ishakhsuvarov committed:
Merge pull request #2617 from nmalevanec/MSI-2616

MSI-2616: Can't create shipment for order with grouped product on custom stock.

duxabilii committed:
Merge branch '2.3-develop' into patch-1

nmalevanec committed:
Shipment Creation MFTF tests modification to support source selection algorithm verification with grouped and configurable product types.

nmalevanec committed:
MSI-2620: Decimal value allowed for input for quantity on source for product
Add validation to qty field depended on qty use decimal field.

nmalevanec committed:
MSI-2620: Decimal value allowed for input for quantity on source for product
Add MFTF test coverage.

duxabilii committed:
Update db_schema.xml

change values which are increased

nmalevanec committed:
MSI-2620: Decimal value allowed for input for quantity on source for product.
Make source default quantity dependant on 'Is Decimal' value.

nmalevanec committed:
MSI-2620: Decimal value allowed for input for quantity on source for product.
Fix MFTF tests.

ishakhsuvarov committed:
Merge pull request #2621 from nmalevanec/MFTF-Modification

Shipment Creation MFTF tests modification to support source selection algorithm verification with grouped and configurable product types.

ishakhsuvarov committed:
Merge pull request #2618 from duxabilii/patch-1

Update module-inventory-distance-based-source-selection/etc/db_schema.xml

ishakhsuvarov committed:
Merge pull request #2623 from nmalevanec/MSI-2620

MSI-2620: Decimal value allowed for input for quantity on source for product

ishakhsuvarov committed:
Adjust for static tests to pass

ishakhsuvarov committed:
Merge pull request #2594 from nmalevanec/MSI-2326

Unskip Web-Api PreventDefaultSourceDisablingTest.

ishakhsuvarov commented on nginx.conf.sample:
```diff
@@ -91,7 +91,15 @@ location ~* ^/update($|/) {
+      fastcgi_param  SCRIPT_FILENAME  $document_root$fastcgi_script_name;
+      include        fastcgi_params;
+  }
+}
```
😢

ishakhsuvarov committed:
Merge branch 'store-pickup' of github.com:magento/inventory into MSI-2287-In-Store-Pickup-orders-placement-from-Admin-Panel

ishakhsuvarov committed:
magento/inventory#2390: Stabilize functional tests

ishakhsuvarov committed:
magento/inventory#2390: Code review fixes

ishakhsuvarov committed:
magento/inventory#2390: Fix error with bundle products orders placement

ishakhsuvarov commented:
@magento run Functional Tests build

ishakhsuvarov commented:
@magento run Functional Tests build

nmalevanec committed:
Merge branch '2.3-develop' into MSI-2287-In-Store-Pickup-orders-placement-from-Admin-Panel

nmalevanec committed:
MFTF test stabilization.

nmalevanec committed:
Static Tests fix.
